Add Quicksight create ingestion Hook and Operator (#21863)
mistercrunch/Airflow: docs/apache-airflow-providers-amazon/operators/quicksight.rst

 .. Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

Amazon QuickSight Operators
========================================

Prerequisite Tasks
------------------

.. include:: _partials/prerequisite_tasks.rst

Overview
--------

> Airflow to Amazon QuickSight integration allows to create and start the SPICE ingestion for dataset.

Reviewer: Might just be me, but "allows to create" feels awkward?  Maybe "allows the user to create" or "allows users" or something?

Author: Done :)

  - :class:`~airflow.providers.amazon.aws.operators.quicksight.QuickSightCreateIngestionOperator`
  - :class:`~airflow.providers.amazon.aws.sensor.quicksight.QuickSightSensor`

Purpose
"""""""

This example DAG ``example_quicksight.py`` uses ``QuickSightCreateIngestionOperator`` for
creating and starting the SPICE ingestion for the dataset configured to use SPICE. In the example,

> we created two ingestion. One of the ingestion waits for the SPICE ingestion to complete while

Reviewer: ```suggestion
we created two ingestions. One of the ingestions waits for the SPICE ingestion to complete while
```

Author: Thanks Done

other ingestion does not wait for completion and uses ``QuickSightSensor`` to check for ingestion
status until it completes

Defining tasks
""""""""""""""

In the following code we create and start a QuickSight SPICE ingestion for the dataset and wait
for its completion.

.. exampleinclude:: /../../airflow/providers/amazon/aws/example_dags/example_quicksight.py

>     :language: python

Reviewer: ```suggestion
    :language: python
    :dedent: 4
```

Something someone showed me recently,. un-indents the code sample for a cleaner look.

Author: Done

    :start-after: [START howto_operator_quicksight]
    :end-before: [END howto_operator_quicksight]

> In the below, we create and start the SPICE ingestion but does not wait for completion. We use

Reviewer: ```suggestion
In the below example, we create and start the SPICE ingestion but do not wait for completion. We use
```

sensor to poll for Ingestion status until it Completes.

.. exampleinclude:: /../../airflow/providers/amazon/aws/example_dags/example_quicksight.py

>     :language: python

Reviewer: ```suggestion
    :language: python
    :dedent: 4
```

Author: Done

    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

Amazon QuickSight Operators
========================================

Prerequisite Tasks
------------------

.. include:: _partials/prerequisite_tasks.rst

Overview
--------

Airflow to Amazon QuickSight integration allows users to create and start the SPICE ingestion for dataset.

  - :class:`~airflow.providers.amazon.aws.operators.quicksight.QuickSightCreateIngestionOperator`
  - :class:`~airflow.providers.amazon.aws.sensor.quicksight.QuickSightSensor`

> Purpose
> """""""
> 
> This example DAG ``example_quicksight.py`` uses ``QuickSightCreateIngestionOperator`` for
> creating and starting the SPICE ingestion for the dataset configured to use SPICE. In the example,
> we created two ingestions. One of the ingestions waits for the SPICE ingestion to complete while
> other ingestion does not wait for completion and uses ``QuickSightSensor`` to check for ingestion
> status until it completes
> 
> 
> Defining tasks
> """"""""""""""
> 
> In the following code we create and start a QuickSight SPICE ingestion for the dataset and wait
> for its completion.
> 
> .. exampleinclude:: /../../airflow/providers/amazon/aws/example_dags/example_quicksight.py
>     :language: python
>     :dedent: 4
>     :start-after: [START howto_operator_quicksight]
>     :end-before: [END howto_operator_quicksight]
> 
> In the below example, we create and start the SPICE ingestion but do not wait for completion. We use
> sensor to poll for Ingestion status until it Completes.
> 
> 
> 
> .. exampleinclude:: /../../airflow/providers/amazon/aws/example_dags/example_quicksight.py
>     :language: python
>     :dedent: 4
>     :start-after: [START howto_operator_quicksight_non_waiting]
>     :end-before: [END howto_operator_quicksight_non_waiting]
> 

Reviewer 2: For the sake of being consistent with other AWS documentations, could you please create 2 sections: one for the operator and one for the sensor

```suggestion
.. _howto/operator:QuickSightCreateIngestionOperator:

Create an Amazon Quicksight ingestion
""""""""""""""""""""""""""""""""""""""""""
Text explaining what the operator does

.. exampleinclude:: /../../airflow/providers/amazon/aws/example_dags/example_quicksight.py
    :language: python
    :dedent: 4
    :start-after: [START howto_operator_quicksight_create_ingestion]
    :end-before: [END howto_operator_quicksight_create_ingestion]

.. _howto/sensor:QuickSightSensor:

 Amazon Quicksight sensor
"""""""""""""""""""""""""""""
Text explaining what the sensor does

.. exampleinclude:: /../../airflow/providers/amazon/aws/example_dags/example_quicksight.py
    :language: python
    :dedent: 4
    :start-after: [START howto_sensor_quicksight]
    :end-before: [END howto_sensor_quicksight]
```
